src/libexpr/parser-state.hh

```diff
@@ -175,6 +180,7 @@ inline Expr * ParserState::stripIndentation(const PosIdx pos,
     /* Figure out the minimum indentation.  Note that by design
        whitespace-only final lines are not taken into account.  (So
        the " " in "\n ''" is ignored, but the " " in "\n foo''" is.) */
+    std::optional<IndentChar> indentChar = std::nullopt;
     bool atStartOfLine = true; /* = seen only whitespace in the current line */
     size_t minIndent = 1000000;
```

Comment:
It is out of scope I think but alright, will do

Reply:
@pennae do you suggest that someone with more than a million indent spaces will be surprised by the parser stripping more spaces than it used to?

Reply:
Agreed with @pennae. This might look silly, but changes like that can have a huge impact, so we shouldn't change behaviour in any way.

Reply:
This particular one seems very silly to me. How likely is it that code exists in the wild that relies on this behavior? We don't have a quantifiable measure of stability, but evaluation results of Nixpkgs stable releases could be a good approximation.

If we take such an absolute stance on compatibility as @pennae and @thufschmitt are suggesting, we can essentially give up on evolving the code. This *can* be a reasonable strategy, but is it desirable? And in any case it should be communicated clearly, otherwise a lot of people will waste their time trying to do things that will end up being rejected.